axelor-account/src/main/java/com/axelor/apps/account/service/invoice/LatePaymentInterestInvoiceServiceImpl.java

```diff
@@ -185,11 +192,37 @@ public Invoice generate() throws AxelorException {
     };
   }
 
-  protected BigDecimal computeLatePaymentInterest(List<InvoiceTerm> invoiceTermList) {
+  protected BigDecimal computeLatePaymentInterest(List<InvoiceTerm> invoiceTermList)
+      throws AxelorException {
     BigDecimal latePaymentAmount = BigDecimal.ZERO;
     for (InvoiceTerm invoiceTerm : invoiceTermList) {
-      latePaymentAmount = latePaymentAmount.add(invoiceTerm.getAmountRemaining());
+      latePaymentAmount = latePaymentAmount.add(computeInterestFromInvoiceTerm(invoiceTerm));
     }
     return latePaymentAmount;
   }
+
+  protected BigDecimal computeInterestFromInvoiceTerm(InvoiceTerm invoiceTerm)
+      throws AxelorException {
+    PaymentMode paymentMode = invoiceTerm.getPaymentMode();
+
+    if (paymentMode == null || paymentMode.getInterestRate() == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_MISSING_FIELD,
+          I18n.get(AccountExceptionMessage.LATE_PAYMENT_INTEREST_NO_PAYMENT_MODE_RATE));
+    }
+    BigDecimal interestRate =
+        paymentMode
+            .getInterestRate()
+            .divide(new BigDecimal("100"), INTEREST_PERCENT_SCALE, RoundingMode.HALF_UP);
```

Comment:
it's better to not round at all to have the most precise amount, it should not cause any issue because rounding is not necessary when dividing per 100.